src/tasks/bundler.rs

```diff
@@ -1,58 +1,62 @@
+use tokio::task::JoinHandle;
+use tokio::time;
+use tracing::{Instrument, debug, trace};
+/// Holds a bundle from the cache with a unique ID and a Zenith bundle.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Bundle {
-    /// The bundle id (a UUID)
+    /// Cache identifier for the bundle.
     pub id: String,
-    /// The bundle itself
-    pub bundle: ZenithEthBundle,
+    /// The corresponding Signet bundle.
+    pub bundle: SignetEthBundle,
 }
 
 /// Response from the tx-pool containing a list of bundles.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct TxPoolBundleResponse {
-    /// the list of bundles
+    /// Bundle responses are available on the bundles property.
     pub bundles: Vec<Bundle>,
 }
 
-/// The BundlePoller polls the tx-pool for bundles and manages the seen bundles.
-#[derive(Debug)]
+/// The BundlePoller polls the tx-pool for bundles.
+#[derive(Debug, Clone)]
 pub struct BundlePoller {
-    /// Configuration
+    /// The builder configuration values.
     pub config: BuilderConfig,
-    /// [`Authenticator`] for fetching OAuth tokens
+    /// Authentication module that periodically fetches and stores auth tokens.
     pub authenticator: Authenticator,
-    /// Already seen bundle UUIDs
-    pub seen_uuids: HashMap<String, Instant>,
+    /// Defines the interval at which the bundler polls the tx-pool for bundles.
+    pub poll_interval_ms: u64,
 }
 
-/// Implements a poller for the block builder to pull bundles from the tx cache.
+/// Implements a poller for the block builder to pull bundles from the tx-pool.
 impl BundlePoller {
     /// Creates a new BundlePoller from the provided builder config.
     pub fn new(config: &BuilderConfig, authenticator: Authenticator) -> Self {
-        Self { config: config.clone(), authenticator, seen_uuids: HashMap::new() }
+        Self { config: config.clone(), authenticator, poll_interval_ms: 1000 }
     }
 
-    /// Fetches bundles from the transaction cache and returns the (oldest? random?) bundle in the cache.
-    pub async fn check_bundle_cache(&mut self) -> eyre::Result<Vec<Bundle>> {
-        let mut unique: Vec<Bundle> = Vec::new();
+    /// Creates a new BundlePoller from the provided builder config and with the specified poll interval in ms.
+    pub fn new_with_poll_interval_ms(
+        config: &BuilderConfig,
+        authenticator: Authenticator,
+        poll_interval_ms: u64,
+    ) -> Self {
+        Self { config: config.clone(), authenticator, poll_interval_ms }
+    }
 
+    /// Fetches bundles from the transaction cache and returns them.
+    pub async fn check_bundle_cache(&mut self) -> eyre::Result<Vec<Bundle>> {
         let bundle_url: Url = Url::parse(&self.config.tx_pool_url)?.join("bundles")?;
         let token = self.authenticator.fetch_oauth_token().await?;
 
-        // Add the token to the request headers
         let result = reqwest::Client::new()
```

Comment:
rather than making a new client each time this is called, let's get https://github.com/init4tech/signet-sdk/pull/39 in and use the `TxCache` struct there

then the cache becomes a prop on `BundlePoller { cach: TxCache, .. }` and we avoid setting this up every time